### ISSUE TITLE
parser module removed in Python 3.10

### ISSUE THREAD

**Issue by johndoe31415** [CDSoft/tpg#1]
Hey there,

I'm the maintainer of llpdf and pdfminify, two of myprojects that happily uses tpg -- it's my favorite parser generator as a matter of fact.

I've just received a curious bug report and looked into it for a second here: https://github.com/johndoe31415/pdfminify/issues/15

It appears that on the system of @andit, tpg cannot locate the parser module. Sure enough, when I try it on my Python3.9 installation:

```
Python 3.9.5 (default, Nov 18 2021, 16:00:48) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import parser
<stdin>:1: DeprecationWarning: The parser module is deprecated and will be removed in future versions of Python
```

It appears this "future version of Python" is 3.10: https://docs.python.org/3/whatsnew/3.10.html indicates

> Removed the parser module, which was deprecated in 3.9 due to the switch to the new PEG parser, as well as all the C source and header files that were only being used by the old parser, including node.h, parser.h, graminit.h and grammar.h.

So it appears tpg won't work with Python 3.10 anymore unless the parser dependency is removed/adapted. :-( Not sure that there's a quick fix, but though I'd let you know.

Thanks again for the great work in TPG.

**Comment by CDSoft** [comment]
Hi,
thanks for your report!
This should be pretty easy to fix. The parser module seems only to be used to check the embedded Python code is valid. One quick fix is to simply remove this check. This will have two effects:

1. the error messages may change
2. the syntax may be checked later, i.e. when the code is actually run. Currently the code is checked when it is generated (e.g. with the tpg script). I think that in general the tpg script is never used and the parser are generated on the fly.

Quick fix coming soon.

**Comment by johndoe31415** [comment]
This sounds awesome, thank you so much!

**Comment by CDSoft** [comment]
Two quick fixes for Python 3.10:

1. parser module removed
2. collections.Callable used only if callable is not defined

The version 3.2.4 should work on Python 3.10 (tested with Python 3.10.2 on Fedora 35 and Python 3.9.7 on Ubuntu 21.10).

It also seems distutils will be removed in Python 3.12. I'll fix it later.
